src/NuGet.Clients/NuGet.PackageManagement.UI/Xamls/PackageManagerControl.xaml.cs

```diff
@@ -361,6 +361,15 @@ private async ValueTask RefreshWhenNotExecutingActionAsync(RefreshOperationSourc
 
         private void EmitRefreshEvent(TimeSpan timeSpan, RefreshOperationSource refreshOperationSource, RefreshOperationStatus status, bool isUIFiltering = false, double? duration = null)
         {
+            NuGetProjectKind projectKind = NuGetProjectKind.Unknown;
+            string projectId = null;
+            if (Model.IsSolution)
+            {
+                IProjectContextInfo project = Model.Context.Projects.First();
+                projectId = project.ProjectId;
+                projectKind = project.ProjectKind;
```

Comment:
Since `projectId` and `projectKind` aren't used for solution level, I think it would be better (lower risk of mistakes) if we did not pass in realistic values, so I'd prefer to use null.

Even better if the other types used (`PackageManagerUIRefreshEvent`) didn't allow mistakes either. For example, you could make the constructor private, then create public static methods `ForSolution` and `ForProject`, where the `ForSolution` version does not contain `projectId` or `projectKind`. Or use different types, one just for solution level PM UI, and another for project level PM UI that extends the solution level event, in order to add the two additional properties.

Reply:
I like the idea of private constructor and factory static methods. Creating different types for the same event could create confusion.

Reply:
Added factory methods.